File: the dashboard for Admin/Models/Compo.cs
using System.ComponentModel;

namespace AQTTan.Models
{
    public class Compo
    {
        public int Id { get; set; }
        [DisplayName("الأسم")]
        public string? Name { get; set; }
        [DisplayName("عدد الأصناف فيه")]
        public int ItemAmount { get; set; }
        [DisplayName("الصورة")]
        public byte[]? Image { get; set; }
    }
}

File: the mobile application/lib/model/compo.dart
import 'dart:convert';
import 'dart:typed_data';

class Compo {
  final int id;
  final String name;
  final int itemAmount; // اجعل هذا الحقل قابل للاحتواء بـ null
  final Uint8List image;

  Compo({
    required this.id,
    required this.name,
    required this.itemAmount, // لم يعد هذا الحقل مطلوبًا
    required this.image,
  });

  factory Compo.fromJson(Map<String, dynamic> json) {
    return Compo(
      id: json['id'], // تعيين القيمة الافتراضية إذا كانت null
      name: json['name'],
      itemAmount: json['itemAmount'], // تعيين القيمة الافتراضية إذا كانت null
      image: base64Decode(json['image']),
    );
  }
}
